rialto_airflow/funders/linker.py

```diff
@@ -0,0 +1,56 @@
+import logging
+from typing import Optional
+
+from sqlalchemy import select
+from sqlalchemy.orm.session import Session
+
+
+from rialto_airflow.database import get_session, Publication, Funder
+from rialto_airflow.funders.dataset import is_federal, is_federal_grid_id
+
+
+def link_publications(snapshot) -> int:
+    count = 0
+    with get_session(snapshot.database_name).begin() as session:
+        stmt = (
+            select(Publication)
+            .where(Publication.dim_json.is_not(None))  # type: ignore
+            .execution_options(yield_per=100)
+        )
+
+        for row in session.execute(stmt):
+            count += 1
+            if count % 100 == 0:
+                logging.info(f"processed {count} publications")
+
+            pub = row[0]
+
+            funders = pub.dim_json.get("funders", [])
+            if funders is None:
+                logging.warning(f"got null funders for {pub.dim_json}")
```

Comment:
Sound like we don't anticipate a lot of these? Was wondering if logging the full dimensions JSON would make the logs grow too large on a full run. Can also wait and see.

Reply:
Yeah, good thought. I'll remove the JSON output.